django_project/cplus_api/api_views/layer.py

```diff
@@ -786,3 +790,88 @@ def post(self, request, *args, **kwargs):
             list(results.values()),
             many=True
         ).data)
+
+
+class ReferenceLayerDownload(APIView):
+    """APIs to fetch and remove layer file."""
+    permission_classes = [AllowAny]
+    authentication_classes = []
+
+    def _stream_response(self, file_path):
+        # Stream the clipped file as a response
+        def file_iterator(fp, chunk_size=8192):
+            with open(fp, 'rb') as file:
+                while chunk := file.read(chunk_size):
+                    yield chunk
+
+        response = StreamingHttpResponse(
+            file_iterator(file_path),
+            content_type="application/octet-stream"
+        )
+        response['Content-Disposition'] = ('attachment; filename'
+                                           '="reference_layer.tif"')
+
+        # Clean up the temporary file after the response is completed
+        response['X-Accel-Buffering'] = 'no'
+        response.close = lambda: os.remove(file_path)
```

Comment:
@zamuzakki if it's clipped file, then you could remove it, but it's the original reference layer, let's keep it to avoid multiple download from minio/s3.

Reply:
@danangmassandy I've added checking to only delete the file if it's the clipped one
https://github.com/kartoza/cplus-api/pull/76/files#diff-b3a029813cb25062e601076866f1e88a0c78745b1d059f3d2b584135b489b2beR821